### PR TITLE
Add more platforms support

### PR DESCRIPTION
**Pull request by tguillem**
Only tested with static build.

### CODE REVIEW

**Comment by vibhoothiiaanand** [comment]
Another way would be trying to use linux-musl when compiling
It works from adb shell the direct binary^^,
For rav1e we are trying to do that way

**Comment by lu-zero** [comment]
For shared libraries we should not put the version in the soname iirc, somebody can confirm it?

@vibhoothiiaanand it is about building for android, not running on android :)

File: src/build_targets.rs [hunk 1]
```diff
@@ -32,12 +32,12 @@ impl BuildTargets {
         let env = &target.env;
 
         let (shared_lib, static_lib, impl_lib, def) = match (os.as_str(), env.as_str()) {
-            ("linux", _) | ("freebsd", _) | ("dragonfly", _) | ("netbsd", _) => {
+            ("linux", _) | ("freebsd", _) | ("dragonfly", _) | ("netbsd", _) | ("android", _) => {
```
Comment (lu-zero) [inline src/build_targets.rs:35]: I confirm you need a different link line, I'll add it now.

**Comment by lu-zero** [comment]
Merged from the other pull requests.

**Pull request closed without merging**